include/drivers/counter.h

```diff
@@ -428,8 +446,12 @@ static inline int counter_cancel_channel_alarm(struct device *dev,
  * @retval -ETIME if @ref COUNTER_TOP_CFG_DONT_RESET was set and new top value
  *		  is smaller than current counter value (counter counting up).
  */
-static inline int counter_set_top_value(struct device *dev,
-					const struct counter_top_cfg *cfg)
+__syscall int counter_set_top_value(struct device *dev,
```

Comment:
same as above, sets callback from top value. Top value callback is optional. Not sure if it's possible to accept user mode code only if callback is null.

Reply:
Same here.